codegen/assets/templates/gocode/store/rdbms/rdbms.go.tpl

```diff
@@ -162,6 +162,25 @@ func (s *Store) Search{{ .expIdentPlural }}(ctx context.Context, {{ template "ex
 	if err != nil {
 		return nil, f, err
 	}
+
+    // @fixme condition per resource
+	if f.IncTotal {
+        // Calc total from the number of items fetched
+        // even if we do build the page navigation
+        f.Total = uint(len(set))
+    }
+    if f.Limit > 0 && uint(len(set)) == f.Limit {
```

Comment:
change this as agreed (move inside the block above and send in proper limit value).